rapidjson.sh

```diff
@@ -7,7 +7,13 @@ tag: v1.1.0
 ---
 #!/bin/sh
 
-cmake $SOURCEDIR                                              \
+GCC_VERSION_MAJOR=$(gcc --version | awk 'NR==1 {print $3}' | cut -d '.' -f1)
+if [ "$GCC_VERSION_MAJOR" -ge "7" ]; then
+  NO_ERROR_FALLTHROUGH="-Wno-error=implicit-fallthrough"
+fi
+
+cmake $SOURCEDIR                                                       \
+      ${NO_ERROR_FALLTHROUGH:+-DCMAKE_CXX_FLAGS=$NO_ERROR_FALLTHROUGH} \
```

Comment:
Mmm... Doesn't this override all the CXXFLAGS, including things like -O2 and co?
 
In any case, this was fixed upstream:

https://github.com/Tencent/rapidjson/commit/cba45fe9de6923b858edb0780e257b7257aa4f7b

why don't we simply pick up a newer version? Can you also fix the linter issue, while we are at it? Thanks.